### PR TITLE
Implement NoteCommands + made minor adjustments to DG

### LINKED ISSUE #110: Add notes feature
Part of #95 
Split by topics or by questions?

### PR DESCRIPTION
**Pull request by joelngyx**
Implemented NoteCommands, made adjustments related to NoteCommand's functionality in other classes (ie UserStorage, Ui), and made adjustments to the User Guide and the Design Guide

Resolves #110

### CODE REVIEW

File: src/main/java/seedu/eduke8/storage/UserStorage.java [hunk 1]
```diff
@@ -101,13 +101,21 @@ private void parseFromTopicJson(JSONObject topic) throws Eduke8Exception {
...
     }
 
+    private void parseFromNoteJson(JSONObject note, NoteList noteList) throws Eduke8Exception {
+        String noteDescription = (String) note.get("description");
+        String text = (String) note.get("text");
+        Note noteObject = new Note(noteDescription, text);
+
+        noteList.add(noteObject);
+    }
+
```
Comment (kstonekuan) [inline src/main/java/seedu/eduke8/storage/UserStorage.java:118]: Well done updating the UserStorage to match your Note changes

File: src/main/java/seedu/eduke8/ui/Ui.java [hunk 2]
```diff
@@ -76,6 +79,21 @@
     private static final String DOT = ".";
     private static final String DOT_SPACE = ". ";
     private static final String DOT_PLURAL = "s.";
+    private static final String ADD_NOTE_PROMPT_FOR_TOPIC = "Enter the topic you would like to add a note to";
```
Comment (kstonekuan) [inline src/main/java/seedu/eduke8/ui/Ui.java:82]: Nice use of constants

File: docs/UserGuide.md [hunk 3]
```diff
@@ -188,7 +191,41 @@ What you should see:
...
+```
+
+### 3.11. Listing out notes: `note list`
+
+Lists out all notes belonging to a specific topic.
+
+Example of usage:
+```
+note list
+```
```
Comment (kstonekuan) [inline docs/UserGuide.md:226]: Good update to the user guide, not sure if these sections should be nested under a notes feature

**Review by kstonekuan — APPROVED** [review]
lgtm

File: src/main/java/seedu/eduke8/ui/Ui.java [hunk 4]
```diff
@@ -198,16 +216,92 @@ public void printTopicList(ArrayList<Displayable> topics) {
...
+                Topic topic = (Topic) topicList.find(topicName);
+                NoteList noteListTopic = topic.getNoteList();
+                ui.printNoteList(noteListTopic);
+            } else {
+                System.out.println(INPUT_ERROR);
+            }
+        } catch (Eduke8Exception e) {
+            ui.printError(e.getMessage());
+        }
+    }
```
Comment (josephhhhhhhhh) [inline src/main/java/seedu/eduke8/ui/Ui.java:291]: Well implemented!

**Pull request merged**